### PR TITLE
Arash/xr table top fix

### PR DESCRIPTION
**Pull request by arash-sh**
**Sample**

New XR tabletop sample 

**Summary**

- Scripts for configuring the tabletop and interacting with it
- Scripts for grabbing the table from distance
- Blueprints for UI, loading the preset locations, and table base  
- Misc. assets

**Tests**

N/A

**ArcGIS Maps SDK Version**
1.6

### CODE REVIEW

File: xr_sample_project/xr_samplesproject.uproject
Comment (mmgaw113) [inline xr_sample_project/xr_samplesproject.uproject:1]: Please revert the changes to this file or make sure the branch is up to date with main.

File: xr_sample_project/Content/Samples/XRTableTop/XRTabletop.png
Comment (mmgaw113) [inline xr_sample_project/Content/Samples/XRTableTop/XRTabletop.png:1]: Did you get rid of the Table that was underneath the map?

Comment (arash-sh) [reply xr_sample_project/Content/Samples/XRTableTop/XRTabletop.png:1]: Yes, based on feedback from Zack, removed to reduce issues with map elevation offset while zooming

Comment (mmgaw113) [reply xr_sample_project/Content/Samples/XRTableTop/XRTabletop.png:1]: Take a look at XR Table top in unity. I am setting the elevation offset every time the location changes through the UI. Based on feedback from adrien and rex (and maybe users), we should have a table rather than just the map floating. Setting the elevation offset everytime makes it so it is never significantly higher than the table and zooming too far doesn't make the map disappear.

Comment (arash-sh) [reply xr_sample_project/Content/Samples/XRTableTop/XRTabletop.png:1]: That's how its being done here as well. But after loading a new location, zooming moves the map up and down and can move the map below the table or too far above it. That's because the elevation offset is being multiplied with the map scale, and as far as I know, that's the same behavior in Unity. 
If the table mesh is absolutely needed, one option would be to have the map's base always be at the same height, taking into account the elevation offset but independent of the zoom, which can cause bigger gaps above the table in mountainous areas.
When/if we add an auto offset feature to the tabletop component, this will not be an issue any longer.

Comment (mmgaw113) [reply xr_sample_project/Content/Samples/XRTableTop/XRTabletop.png:1]: Gotcha, I do remember that being a problem where the map would sometimes go in or below the table.

File: xr_sample_project/Config/DefaultEngine.ini
Comment (arash-sh) [inline xr_sample_project/Config/DefaultEngine.ini:1]: Re-enabled anti-aliasing to avoid flickering on the map and table

File: xr_sample_project/Source/xr_samplesproject/XRTableTop/XRTabletopComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+bool UXRTabletopComponent::FindArcGISCameraLocationInHierarchy()
+{
+	ArcGISCameraLocation = nullptr;
+	TArray<AActor*> actors;
+	GetOwner()->GetAttachedActors(actors, true, true);
+	for (auto a : actors)
+	{
+		if (a->FindComponentByClass<UArcGISCameraComponent>() && a->FindComponentByClass<UArcGISLocationComponent>())
+			ArcGISCameraLocation = a->FindComponentByClass<UArcGISLocationComponent>();
```
Comment (ZackAllen) [inline xr_sample_project/Source/xr_samplesproject/XRTableTop/XRTabletopComponent.cpp:357]:
```suggestion
		if (a->FindComponentByClass<UArcGISCameraComponent>() && a->FindComponentByClass<UArcGISLocationComponent>())
		{
			ArcGISCameraLocation = a->FindComponentByClass<UArcGISLocationComponent>();
		}
```

Comment (arash-sh) [reply xr_sample_project/Source/xr_samplesproject/XRTableTop/XRTabletopComponent.cpp:357]: Good catch!

**Review by ZackAllen — APPROVED** [review]

**Review by avr757 — APPROVED** [review]
